beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/forward/multipeer/chains/ThrottlingSyncSource.java

```diff
@@ -56,9 +56,17 @@ public SafeFuture<Void> requestBlocksByRange(
       final UInt64 startSlot,
       final UInt64 count,
       final RpcResponseListener<SignedBeaconBlock> listener) {
-    if (blocksRateTracker.wantToRequestObjects(count.longValue()) > 0) {
-      LOG.debug("Sending request for {} blocks", count);
-      return delegate.requestBlocksByRange(startSlot, count, listener);
+    final UInt64 updatedCount =
+        count.min(UInt64.valueOf(blocksRateTracker.allowedRequestObjectsCount(count.longValue())));
```

Comment:
Wouldn't it be better to update `RateTracker` implementation to return min of requested and remaining requests?

Reply:
Actually it already does that. I refactored the code, there were an unnecessary double `min` operation...